app/ui/src/app/integration/edit-page/step-configure/data-mapper/data-mapper-host.component.ts

```diff
@@ -99,54 +99,20 @@ export class DataMapperHostComponent implements OnInit {
 
     // Populate all supported DataShape from previous DataShape aware steps as source documents
     for (const pair of previousSteps) {
-      const dataShapeToFetch = pair.step.action.descriptor.outputDataShape;
-      if (this.isSupportedDataShape(dataShapeToFetch)) {
-        this.addInitializationTask();
-        this.currentFlowService.fetchOutputDataShapeFor(pair.step).then(dataShape => {
-          this.addSourceDocument(pair.step.id, pair.index, dataShape);
-          this.removeInitializationTask();
-        })
-        .catch(response => {
-          this.cfg.errorService.error(
-            'Failed to load source data type for Step "' + pair.step.name + '": '
-            + ' Unsupported data type "' + dataShapeToFetch + '": ' + response, '');
-          this.removeInitializationTask();
-        });
+      const outputDataShape = pair.step.action.descriptor.outputDataShape;
+      if (this.isSupportedDataShape(outputDataShape)) {
+        this.addSourceDocument(pair.step.id, pair.index, outputDataShape);
       }
     }
 
     // Next DataShape aware step must have a supported input DataShape, which describes a target document
     const targetPair = this.currentFlowService.getSubsequentStepsWithDataShape(this.position)[1];
-    this.addInitializationTask();
-    this.currentFlowService.fetchInputDataShapeFor(targetPair.step).then(dataShape => {
-      if (!this.addTargetDocument(targetPair.step.id, targetPair.index, dataShape)) {
-        this.cfg.errorService.error(
-          'Unsupported target data type for Step "' + targetPair.step.name + '": ' + dataShape, '');
-        this.removeInitializationTask();
-        return;
-      }
-      this.currentFlowService.events.emit({
```

Comment:
I think we still need to fire this event, this is to copy data shapes into datamapper step from previous/subsequent steps.

Reply:
ah, okay, I'll rework that, thanks!